### PR TITLE
Add index filtering via MIQ Common Gem

### PR DESCRIPTION
**Pull request by lindgrenj6**
https://projects.engineering.redhat.com/browse/SSP-307

Adds filtering on the index methods via ManageIQ::API::Common::Filter.

- [x] fix tests
- [x] add docs https://github.com/ManageIQ/manageiq-api-common/pull/57

### CODE REVIEW

**Comment by lindgrenj6** [comment]
@miq-bot add_reviewer bzwei 
@miq-bot add_reviewer hsong-rh

File: public/approval/v1.0/openapi.json [hunk 1]
```diff
@@ -1050,27 +1076,27 @@
...
-                                "type": "string",
-                                "format": "date-time",
-                                "description": "Timestamp of creation"
-                            },
-                            "stage_id": {
-                                "type": "string",
-                                "description": "Associated stage id"
-                            }
-                        }
+                        "type": "object"
```
Comment (hsong-rh) [inline public/approval/v1.0/openapi.json]: @lindgrenj6 what's the reason you made these changes? reformat?

Comment (lindgrenj6) [reply public/approval/v1.0/openapi.json]: Yes, when defining the `Out` objects initially it looks like we used `allOf` which includes fields, and then after that declared the extra fields. That _can_ go in the same block as `allOf` but that makes it look like the the object has no fields. So all I did was move the extra field declarations outside of the `allOf` part, so they show up the way the `Api::Doc` parser is expecting.

Comment (hsong-rh) [reply public/approval/v1.0/openapi.json]: 👍

**Pull request merged**